File: R/functions/createGradient.R
createGradient <- function(S, Free, M) {
  
  for (i in 1:S$n_items) {
    
    M$gradient[i,] <- (Free$alpha - (0:(S$n_items-1)) * Free$gamma) * Free$delta^(i-1)
    
  }
  
  return(M)
  
}

File: Julia/functions/createGradient.jl
function createGradient(S, F, M)

    # create gradient for each output position at the same time, resulting in a n-by-n matrix
    for i in 1:S.n_items
        M.gradient[:,i] .= collect(F.alpha .- (0.0:Float64(S.n_items - 1)) .* F.gamma) * F.delta^Float64(i-1)
    end

    return nothing
    
end